app/models/parameter.rb

```diff
@@ -67,14 +68,19 @@ def hash_for_include_source(source, source_name = nil)
     options = {
       :value => value, :source => source, :key_type => key_type,
       :safe_value => safe_value, :parameter_type => parameter_type,
-      :hidden_value? => hidden_value?
+      :hidden_value? => hidden_value?,
+      :searchable_value => searchable_value
     }
     options[:source_name] = source_name if source_name
     options
   end
 
   private
 
+  def set_searchable_value
+    self.searchable_value = Parameter.format_value_before_type_cast(value, key_type)
```

Comment:
Do we want yaml saved as yaml?
This is pretty tricky question I guess... I would say we would be fine with json, what would be searchable more easily, but we would loose some features of yaml, those are not usefull for searching, so it would probably be better format all the composed formats as json...
Only fallback would be, that I would imagine in the future droping the yaml serialized column.